DuckDuckGo/DaxDialogs.swift

```diff
@@ -257,8 +265,19 @@ final class DaxDialogs {
         settings.fireButtonEducationShownOrExpired = true
         return ActionSheetSpec.fireButtonEducation
     }
+    
+    func nextBrowsingMessageIfShouldShow(for privacyInfo: PrivacyInfo) -> BrowsingSpec? {
```

Comment:
I assume this function is supposed to work like the counter part for the home dialogs `peekNextHomeScreenMessage()`?

The only difference I see and maybe it is a problem that all browsing message returning function calls from `nextBrowsingMessage(privacyInfo: PrivacyInfo)` like:
- `searchMessage()`
- `majorTrackerMessage(host)`
- `majorTrackerOwnerMessage(host, owner)`
-  `trackersBlockedMessage(entityNames)` 
-  `noTrackersMessage()`
they all change the state (boolean flags in `settings: DaxDialogsSettings` of what dialogs were already seen). I think it may cause the issue that if we presented one dialog and we attempt to present some other (due to change state) it will be suppressed but not only on this website - the flags will reflect the state as it was already seen by the user hence we will never present it again. I see this as an issue especially as it is the blocked trackers educational Dax Dialog like in the reported YT case.

Reply:
I'm not sure I follow. The idea was for it to work slightly differently to peekNextHomeScreenMessage to avoid having to duplicate any logic and make things easier and reduce the chance of introducing new bugs if we ever want to change the order of dax dialogs or anything like that in future. I also wanted to avoid changing the logic of nextBrowsingMessage as much as possible since it's reasonably involved and would be easy to accidentally introduce a new bug.

My intention was for nextBrowsingMessage() to never be called for the new case where we don't intend to show a dialog (i.e. when still on the same page), so that the state isn't modified again, and we should avoid the problem I think you're describing. But I may have either misunderstood what you're saying, or failed to deliver on that intention

Reply:
Sorry, I was me that missed how it was called and guarded. I think you make a good point - no new message will be requested if the has not changed hence the state is coherent.